### PR TITLE
Moves relay classes to a new package

### PR DESCRIPTION
**Pull request by greenart7c3**

### CODE REVIEW

**Comment by vitorpamplona** [comment]
Nice!

**Comment by vitorpamplona** [comment]
Data Sources as well? Aren't they tightly coupled with Amethyst's UI? I never thought anyone else could use them.

**Comment by vitorpamplona** [comment]
ohh just the base class. That makes sense.

File: app/src/main/java/com/vitorpamplona/amethyst/LocalPreferences.kt [hunk 1]
```diff
@@ -75,7 +75,7 @@ private object PrefKeys {
     const val SAVED_ACCOUNTS = "all_saved_accounts"
     const val NOSTR_PRIVKEY = "nostr_privkey"
     const val NOSTR_PUBKEY = "nostr_pubkey"
-    const val RELAYS = "relays"
+    const val RELAYS = "com/vitorpamplona/ammolite/relays/relays"
```
Comment (vitorpamplona) [inline app/src/main/java/com/vitorpamplona/amethyst/LocalPreferences.kt:78]: check this

**Comment by greenart7c3** [comment]
> ohh just the base class. That makes sense.

Yes, i also added a few functions to the listeners classes that I use in Amber
And fixed a bug when using the sendToSelectedRelays function and the relay is not ready

**Comment by vitorpamplona** [comment]
Very cool!

**Comment by davotoula** [comment]
Is this related to a feature or an issue?

What's the context for this change?

Thanks!

**Comment by vitorpamplona** [comment]
It was a long requested feature to break the base layers of the app into multiple libraries. We can keep pushing things into libraries to help others develop. 

In this case, I think @greenart7c3 needed some of the relay classes for Amber/Citrine.

**Pull request merged**